packages/zapp/console/src/components/flytegraph/ReactFlow/transformDAGToReactFlowV2.tsx

```diff
@@ -84,9 +85,13 @@ const buildReactFlowDataProps = (props: BuildDataProps) => {
 
   const nodeLogsByPhase: LogsByPhase = nodeExecutionsById?.[node.scopedId]?.logsByPhase;
 
-  const cacheStatus: CatalogCacheStatus =
-    nodeExecutionsById?.[scopedId]?.closure.taskNodeMetadata?.cacheStatus ??
-    CatalogCacheStatus.CACHE_DISABLED;
+  // get the cache status for mapped task
+  const isMapCache =
+    taskType === TaskType.ARRAY && nodeValue?.template?.metadata?.cacheSerializable;
```

Comment:
Same here, use i`sMapTaskType(taskType)` check instead of comparing to `TaskType.ARRAY`